frontend/src/metabase/query_builder/components/ActionsWidget.jsx

```diff
@@ -88,7 +88,11 @@ export default class ActionsWidget extends Component {
         const action = getModeActions(mode, card, tableMetadata)[index];
         if (action && action.popover) {
             this.setState({ selectedActionIndex: index });
-        } else if (action && action.card) {
+        } else if (action && action.url) {
+          const nextUrl = action.url();
+          this.props.onChangeLocation(nextUrl);
```

Comment:
Need to add `onChangeLocation` to the `Props` type but otherwise this seems reasonable.

Also a null check and `MetabaseAnalytics.trackEvent` like we have for `action.card` would be good.